drivers/remote/driver.go

```diff
@@ -28,19 +29,36 @@ func newDriver(name string, client *plugins.Client) driverapi.Driver {
 
 // Init makes sure a remote driver is registered when a network driver
 // plugin is activated.
-func Init(dc driverapi.DriverCallback, config map[string]interface{}) error {
-	plugins.Handle(driverapi.NetworkPluginEndpointType, func(name string, client *plugins.Client) {
-		// negotiate driver capability with client
-		d := newDriver(name, client)
-		c, err := d.(*driver).getCapabilities()
-		if err != nil {
-			log.Errorf("error getting capability for %s due to %v", name, err)
-			return
-		}
-		if err = dc.RegisterDriver(name, d, *c); err != nil {
-			log.Errorf("error registering driver for %s due to %v", name, err)
-		}
-	})
+func Init(dc driverapi.DriverCallback, ps interface{}, config map[string]interface{}) error {
+	// Unit test code is unaware of a true PluginStore. So we fall back to v1 plugins.
+	if ps != nil {
+		ps.(plugingetter.PluginGetter).Handle(driverapi.NetworkPluginEndpointType, func(name string, client *plugins.Client) {
+			// negotiate driver capability with client
+			d := newDriver(name, client)
+			c, err := d.(*driver).getCapabilities()
+			if err != nil {
+				log.Errorf("error getting capability for %s due to %v", name, err)
+				return
+			}
+			if err = dc.RegisterDriver(name, d, *c); err != nil {
+				log.Errorf("error registering driver for %s due to %v", name, err)
+			}
+		})
+	} else {
+		plugins.Handle(driverapi.NetworkPluginEndpointType, func(name string, client *plugins.Client) {
```

Comment:
The callback function seems identical. Can you pls reuse it instead of duplicating it here ?